src/node_code_integrity.cc

```diff
@@ -0,0 +1,237 @@
+#include "node_code_integrity.h"
+#include "env-inl.h"
+#include "node.h"
+#include "node_external_reference.h"
+#include "v8.h"
+
+namespace node {
+
+using v8::Boolean;
+using v8::Context;
+using v8::FunctionCallbackInfo;
+using v8::Local;
+using v8::Object;
+using v8::Value;
+
+namespace per_process {
+bool isWldpInitialized = false;
+pfnWldpCanExecuteFile WldpCanExecuteFile;
+pfnWldpGetApplicationSettingBoolean WldpGetApplicationSettingBoolean;
+pfnWldpQuerySecurityPolicy WldpQuerySecurityPolicy;
+}  // namespace per_process
+
+namespace code_integrity {
+
+static PCWSTR NODEJS = L"Node.js";
+static PCWSTR ENFORCE_CODE_INTEGRITY_SETTING_NAME = L"EnforceCodeIntegrity";
+static PCWSTR DISABLE_INTERPRETIVE_MODE_SETTING_NAME =
+    L"DisableInteractiveMode";
+
+void InitWldp(Environment* env) {
+  if (per_process::isWldpInitialized) {
+    return;
+  }
+
+  HMODULE wldp_module =
+      LoadLibraryExA("wldp.dll", nullptr, LOAD_LIBRARY_SEARCH_SYSTEM32);
+
+  if (wldp_module == nullptr) {
+    return env->ThrowError("Unable to load wldp.dll");
+  }
+
+  per_process::WldpCanExecuteFile =
+      (pfnWldpCanExecuteFile)GetProcAddress(wldp_module, "WldpCanExecuteFile");
+
+  per_process::WldpGetApplicationSettingBoolean =
+      (pfnWldpGetApplicationSettingBoolean)GetProcAddress(
+          wldp_module, "WldpGetApplicationSettingBoolean");
+
+  per_process::WldpQuerySecurityPolicy =
+      (pfnWldpQuerySecurityPolicy)GetProcAddress(wldp_module,
+                                                 "WldpQuerySecurityPolicy");
+
+  per_process::isWldpInitialized = true;
+}
+
+static void IsFileTrustedBySystemCodeIntegrityPolicy(
+    const FunctionCallbackInfo<Value>& args) {
+  CHECK_EQ(args.Length(), 1);
+  CHECK(args[0]->IsString());
+
+  Environment* env = Environment::GetCurrent(args);
+  if (!per_process::isWldpInitialized) {
+    InitWldp(env);
+  }
+
+  BufferValue path(env->isolate(), args[0]);
+  if (*path == nullptr) {
+    return env->ThrowError("path cannot be empty");
+  }
+
+  HANDLE hFile = CreateFileA(*path,
+                             GENERIC_READ,
+                             FILE_SHARE_READ,
+                             nullptr,
+                             OPEN_EXISTING,
+                             FILE_ATTRIBUTE_NORMAL,
+                             nullptr);
+
+  if (hFile == INVALID_HANDLE_VALUE || hFile == nullptr) {
+    return env->ThrowError("Unable to open file");
+  }
```

Comment:
You need to throw appropriate error here. All errors in Node.js needs to have code property. Please take a look at errors.md for the correct error message.